api/api.go

```diff
@@ -730,9 +739,104 @@ func OrbSource(ctx context.Context, logger *logger.Logger, namespace string, orb
 	return response.Orb.Versions[0].Source, nil
 }
 
+// ListOrbs queries the API to find all orbs.
+// Returns a collection of Orb objects containing their relevant data. Logs
+// request and parse errors to the supplied logger.
+func ListOrbs(ctx context.Context, logger *logger.Logger) ([]Orb, error) {
+	// Define a structure that matches the result of the GQL
+	// query, so that we can use mapstructure to convert from
+	// nested maps to a strongly typed struct.
+	type orbList struct {
+		Orbs struct {
+			TotalCount int
+			Edges      []struct {
+				Cursor string
+				Node   struct {
+					Name     string
+					Versions []struct {
+						Version string
+						Source  string
+					}
+				}
+			}
+			PageInfo struct {
+				HasNextPage bool
+			}
+		}
+	}
+
+	request := graphql.NewRequest(`
```

Comment:
There is an existing pattern for queries:
https://github.com/CircleCI-Public/circleci-cli/blob/84650275c632be0dbc6cab4cfab345d4ae17ce46/api/api.go#L866

We've been using our `client` library to initiate requests -- I refactored the code for listing namespace orbs to do so previously.